improver/blending/spatial_weights.py

```diff
@@ -289,15 +150,150 @@ def create_template_slice(cube_to_collapse, blend_coord):
+                rescaled_weights.append(weights_slice)
+                is_rescaled.append(
+                    weights_slice.copy(data=np.zeros(weights_slice.shape, dtype=int))
+                )
+            else:
+                weights_orig = weights_slice.data.copy()
+
+                # calculate the distance to the nearest invalid point, in grid squares,
+                # for each point on the grid
+                distance = distance_transform_edt(weights_nonzero)
+
+                # calculate a 0-1 scaling factor based on the distance from the
+                # nearest invalid data point, which scales between 1 at the fuzzy length
+                # towards 0 for points closest to the edge of the mask
+                fuzzy_factor = rescale(
+                    distance, data_range=[0.0, self.fuzzy_length], clip=True
+                )
+
+                # multiply existing weights by fuzzy scaling factor
+                rescaled_weights_data = np.multiply(
+                    weights_slice.data, fuzzy_factor
+                ).astype(FLOAT_DTYPE)
+                rescaled_weights.append(weights_slice.copy(data=rescaled_weights_data))
+
+                # identify spatial points where weights have been rescaled
+                is_rescaled_data = np.where(
+                    rescaled_weights_data != weights_orig, 1, 0
+                ).astype(int)
+                is_rescaled.append(weights_slice.copy(data=is_rescaled_data))
+
+        weights = rescaled_weights.merge_cube()
+        rescaled = is_rescaled.merge_cube()
+
+        return weights, rescaled
+
+    def _rescale_unmasked_weights(self, weights, is_rescaled):
+        """Increase weights of unmasked slices at locations where masked slices
+        have been smoothed, so that the sum of weights over self.blend_coord is
+        re-normalised (sums to 1) at each point and the relative weightings of
+        multiple unmasked slices are preserved.  Modifies weights cube in place.
+
+        Args:
+            weights (iris.cube.Cube):
+                Cube of weights to which fuzzy smoothing has been applied to any
+                masked slices
+            is_rescaled (iris.cube.Cube):
+                Cube matching weights.shape, with value of 1 where masked weights
+                have been rescaled, and 0 where they are unchanged.
+        """
+        rescaled_data = np.multiply(weights.data, is_rescaled.data)
+        unscaled_data = np.multiply(weights.data, (1 - is_rescaled.data))
+        unscaled_sum = np.sum(unscaled_data, axis=self.blend_axis)
+        required_sum = 1.0 - np.sum(rescaled_data, axis=self.blend_axis)
+        normalisation_factor = np.where(
+            unscaled_sum > 0, np.divide(required_sum, unscaled_sum), 0
+        )
+        normalised_weights = (
+            np.multiply(unscaled_data, normalisation_factor) + rescaled_data
+        )
+        weights.data = normalised_weights.astype(FLOAT_DTYPE)
```

Comment:
Is the casting required? `weights.data` should always be in the correct dtype as this is a private function

Reply:
`np.multiply` escalates the datatype, for reasons that I don't understand (since all the inputs should have a datatype of 32-bit precision).

Reply:
That's very strange, it should only scale to the highest dtype, but I have to be honest, I've never checked it with np.multiply

Reply:
Turns out there's actually an argument to `np.multiply` to enforce a result dtype.  However given what's happening in this function, it's actually tidier just to cast the result at the end.